ckanext/harvest/harvesters/ckanharvester.py

```diff
@@ -36,17 +41,34 @@ def _get_content(self, url):
         api_key = self.config.get('api_key',None)
         if api_key:
             http_request.add_header('Authorization',api_key)
-        http_response = urllib2.urlopen(http_request)
 
+        try:
+            http_response = urllib2.urlopen(http_request)
+        except urllib2.HTTPError, e:
+            raise ContentFetchError(
+                'Could not fetch url: %s, HTTP error code: %s' % 
+                (url, e.code)
+            )
```

Comment:
I believe this also needs to catch urllib2.URLError, in cases such as where the url has bad syntax, or the socket connection times out.